### ISSUE TITLE
Validate AWS bucket names

### ISSUE THREAD

**Issue by maxtheman** [Miserlou/Zappa#689]
## Context
I was attempting to run `zappa deploy dev` for the first time on a new project. This is my first experience with zappa.

## Expected Behavior
My app was the "hello world" code from Flask's main page, and so I expected it to deploy correctly.

## Actual Behavior
Boto3 threw an error saying:

`ClientError: An error occurred (InvalidBucketName) when calling the CreateBucket operation: The specified bucket is not valid.`

## Fix
The problem was that I had an underscore "_" in my bucket name in zappa_settings.json

Once I removed that underscore it worked fine.

I think the problem here is that Zappa doesn't warn you what's going on -- the bucket creation fails silently and doesn't warn the user

## Steps to Reproduce
1. Put unsupported character in zappa name
2. Run a deploy

## Your Environment

* Zappa version used: Most recent
* Operating System and Python version: Mac OS, python 3
* Link to your project (optional):
* Your `zappa_settings.py`: 

OLD:
`{
    "dev": {
        "app_function": "ux_app.app",
        "s3_bucket": "ux_app"
    }
}`

NEW:

`{
    "dev": {
        "app_function": "ux_app.app",
        "s3_bucket": "ux-app"
    }
}`

Now that it's deployed I am super impressed. Thank you for building this!

**Comment by KoreyPeters** [comment]
Maybe I'm misunderstanding but I feel like it did tell you. Boto3 tried to validate the name by creating it, and when it couldn't it threw the stack trace you pasted in which says the bucket name was invalid.

I'm new to Zappa and think it's pretty cool too. :)

**Comment by RichardCullen** [comment]
I agree - Zappa CLI should check for a valid bucket name on data entry. Saves failed deployments and head scratching.
